lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1590,30 +1591,53 @@ void hlsl::SerializeDxilContainerForModule(DxilModule *pModule,
       // If the debug name should be specific to the sources, base the name on the debug
       // bitcode, which will include the source references, line numbers, etc. Otherwise,
       // do it exclusively on the target shader bitcode.
+
       pHashStream = (int)(Flags & SerializeDxilFlags::DebugNameDependOnSource)
                         ? CComPtr<AbstractMemoryStream>(pModuleBitcode)
                         : CComPtr<AbstractMemoryStream>(pProgramStream);
+
+      // Use user specified debug name if a) it's given and b) it's not a path
+      bool UseDebugName = DebugName.size() && !DebugName.endswith(llvm::StringRef("\\"));
```

Comment:
Trailing backslashes are not usually significant / used to identify directories.

Reply:
That's how dxc works right now. If the argument you give to /Fd ends with a backslash, it'll assume it's a directory and put a debug info file with a generated name in there.